ratelimits/limit.go

```diff
@@ -0,0 +1,105 @@
+package ratelimits
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/letsencrypt/boulder/config"
+	"github.com/letsencrypt/boulder/strictyaml"
+)
+
+type limit struct {
+	// Burst specifies maximum concurrent allowed requests at any given time. It
+	// must be greater than zero.
+	Burst int64
+
+	// Count is the number of requests allowed per period. It must be greater
+	// than zero.
+	Count int64
+
+	// Period is the duration of time in which the count (of requests) is
+	// allowed. It must be greater than zero.
+	Period config.Duration
+}
+
+type limits map[string]limit
+
+func parseDefaultName(k string) (string, error) {
+	name, ok := stringToName[k]
+	if !ok {
+		return "", fmt.Errorf(
+			"unrecognized limit %q, must be one of %v", k, limitNames)
+	}
+	return nameToEnumString(name), nil
+}
+
+func parseOverrideNameId(k string) (string, string, error) {
+	nameAndId := strings.SplitN(k, ":", 2)
+	nameStr := nameAndId[0]
+	if nameStr == "" {
+		return "", "", fmt.Errorf("empty name in override %q, must be formatted 'name:id'", k)
+	}
```

Comment:
Given the comment above, I'd consider removing this check, and just letting it be handled by the empty string being an unrecognized name on line 48.